pkg/xray/resource_xray_settings.go

```diff
@@ -0,0 +1,88 @@
+package xray
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func resourceXraySettings() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceXrayDbSyncTimeUpdate,
+		ReadContext:   resourceXrayDbSyncTimeRead,
+		UpdateContext: resourceXrayDbSyncTimeUpdate,
+		DeleteContext: resourceXrayDbSyncTimeDelete,
+		Description:   "Provides an Xray DB Sync Time resource.",
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"db_sync_updates_time": {
+				Type:             schema.TypeString,
+				Required:         true,
+				Description:      "The time of the Xray DB sync daily update job. Format HH:mm",
+				ValidateDiagFunc: matchesHoursMinutesTime,
+			},
+		},
+	}
+}
+
+type DbSyncDailyUpdatesTime struct {
+	DbSyncTime string `json:"db_sync_updates_time"`
+}
+
+func unpackDBSyncTime(s *schema.ResourceData) DbSyncDailyUpdatesTime {
+	d := &ResourceData{s}
+	dbSyncTime := DbSyncDailyUpdatesTime{
+		DbSyncTime: d.getString("db_sync_updates_time", false),
+	}
+	return dbSyncTime
+}
+
+func packDBSyncTime(dbSyncTime DbSyncDailyUpdatesTime, d *schema.ResourceData) diag.Diagnostics {
+	if err := d.Set("db_sync_updates_time", dbSyncTime.DbSyncTime); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceXrayDbSyncTimeRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	dbSyncTime := DbSyncDailyUpdatesTime{}
+	resp, err := m.(*resty.Client).R().SetResult(&dbSyncTime).Get("xray/api/v1/configuration/dbsync/time")
+	if err != nil {
+		if resp != nil && resp.StatusCode() != http.StatusOK {
+			log.Printf("Critical error. DB sync settings (%s) not found.", d.Id())
+		}
+		return diag.FromErr(err)
+	}
+	packDBSyncTime(dbSyncTime, d)
+	return nil
+}
+
+func resourceXrayDbSyncTimeUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	dbSyncTime := unpackDBSyncTime(d)
+	resp, err := m.(*resty.Client).R().SetBody(dbSyncTime).Put("xray/api/v1/configuration/dbsync/time")
+	if err != nil {
+		if resp != nil && resp.StatusCode() == http.StatusNotFound {
+			log.Printf("[WARN] DB sync settings (%s) not found, removing from state", d.Id())
```

Comment:
This is still an error - if you can't actually delete this resource then it should be absolutely impossible to get 404. This is an error

Reply:
Updated to reject H:mm

Reply:
Do I remove it from the state on update error? Regex verification won't allow API call to error out on the wrong time format. So the only possibility, as you said, it a very broken installation.

Reply:
fixed